### ISSUE TITLE
Download stopped at 31% speed is 0B/s

### ISSUE THREAD

**Issue by vedogz098** [blastrock/pkgj#103]
Version of firmware: [3.67]
Version of PKGj: [0.22]

## Steps to reproduce the issue

- I tried to download some other game first, like Dragon Quest Builders and it succeeded
- Then I started Disgaea 4 download
- I made sure to tap the screen every 10 mins or so even if my standby is after 30 mins just to make sure it doesn't sleep
- After waiting for some time it just stopped
- But note that I am not the only one using the wifi at the time and almost all the others are watching videos online
- No errors popping out

Obtained result: Downloading stopped at 31% with 0 B/s speed
Expected result: I should have completed

Game title ID: PCSE00360
Disgaea 4: A Promise Revisited (USA)

**Comment by blastrock** [comment]
I think this is a problem with your internet, and, as many people are having this kind of problem, maybe it's just the vita that is very picky with internet connections.

I will probably implement download resuming to allow you recovering your download and not restarting it from scratch.

**Comment by vedogz098** [comment]
Resuming download will be very much welcome. By the way, I also saw this from another issue post but it looks like the download is continuing. It is just the speed and progress that is stucked. It is stopping at the approximately 1.5gb downloaded content part of the download.

**Comment by vedogz098** [comment]
Also, I noticed that when I press power button to let the Vita go on standby it will cause a download error. This does not happen when I let the Vita automatically go to standby after some time (mine is 30 mins).

**Comment by blastrock** [comment]
Yes, I don't know of any way to keep the vita awake. It seems that letting it go to sleep is ok, but putting it to sleep is not.

**Comment by ronnyc4** [comment]
vedogz098 "it looks like the download is continuing. It is just the speed and progress that is stucked."
the progress bar stuck, its show (0 B/s, 19%) but it seems like it keeps downloading cause my internet speed keep running.
is it actually downloading the game or it just download dummy file that keep deleting itself, download it again than repeat.

**Comment by vedogz098** [comment]
It actually completes the download. I just leave my downloads overnight and they are completed once I wake up. It may not be a big issue but it still will be great if the progress bar will work okay. Anyways I am thankful for this.

**Comment by blastrock** [comment]
Maybe related to #112 for which I gave an explanation.

**Comment by cbecerrat** [comment]
I had this same issue "Download stopped at 31% speed is 0B/s", the download stopped at any pertentage, though. It happened with big games (bigger than 3Gb). But with a little of patience, I realized pkgj continues downloading the game, as I saw how memory space continued decreasing after a while. By waiting enough time, the download completes and install the game correctly, even if it was stoped at certain percentage for a long time. I had this behaviour on v0.23 and 0.24. I haven't tried newer versions as my memory was corrupted due to being a fake microsd. Will test v0.27 soon. I had this behaviour downloading Final Fantasy X HD and World of Final Fantasy.
